File: task-03/Ruby/subtask-03.rb
def print_star_pattern(n)
  (1..n).step(2) do |i|
    puts ' ' * ((n - i) / 2) + '*' * i
  end

  (n-2).step(1, -2) do |i|
    puts ' ' * ((n - i) / 2) + '*' * i
  end
end

print "Enter a number: "
n = gets.to_i
print_star_pattern(n)

File: task-03/Elixir/subtask-03.ex
defmodule StarPattern do
  def print_star_pattern(n) do
    # Use a range with a step value directly after the range definition
    for i <- 1..n//1 do
      if rem(i, 2) == 1 do
        stars = String.duplicate("*", i)
        padding = String.duplicate(" ", div(n - i, 2))
        IO.puts("#{padding}#{stars}")
      end
    end

    for i <- (n - 2)..1//-1 do
      if rem(i, 2) == 1 do
        stars = String.duplicate("*", i)
        padding = String.duplicate(" ", div(n - i, 2))
        IO.puts("#{padding}#{stars}")
      end
    end
  end
end

IO.write "Enter a number: "
n = IO.gets("") |> String.trim() |> String.to_integer()
StarPattern.print_star_pattern(n)